node/engine/messageservice/p2p-message-service/service.go

```diff
@@ -170,19 +172,43 @@ func (ms *P2PMessageService) setupDht(bootPeers []string) {
 				ms.logger.Debug().Msgf("peer info: %v", peer)
 			}
 
-			// Once we've discovered all bootPeers, stop the ticker
+			// Once we've connected to enough peers, stop the ticker
 			if actualPeers >= expectedPeers {
-				ms.logger.Debug().Msgf("discovered all expected bootPeers")
+				ms.logger.Info().Msgf("initial threshold for peer connections has been met")
 				ticker.Stop()
 				break
 			}
 		}
+
+		// Add my state channel address to the custom dht
+		recordData := &DhtData{
+			SCAddr:    ms.me.String(),
+			PeerID:    ms.Id().String(),
+			Timestamp: time.Time.Unix(time.Now()),
+		}
+		recordDataBytes, err := json.Marshal(recordData)
+		ms.checkError(err)
+
+		signature, err := ms.key.Sign(recordDataBytes)
+		ms.checkError(err)
+
+		fullRecord := &DhtRecord{
+			Data:      *recordData,
+			Signature: signature,
+		}
+		fullRecordBytes, err := json.Marshal(fullRecord)
+		ms.checkError(err)
+
+		key := DHT_RECORD_PREFIX + ms.me.String()
+		err = ms.dht.PutValue(ctx, key, fullRecordBytes)
+		ms.checkError(err)
+		ms.logger.Info().Msgf("Added value to dht - [key: %v, value: %v]", key, ms.Id())
```

Comment:
nit but I think if we do this it makes the logs easier to parse:
```suggestion
			ms.logger.Info().Str(key, ms.Id().Pretty()).Msg("Added value to dht")
```